### PR TITLE
Move quic_logging_impl.{h,cc} to quiche_logging_impl.{h,cc}

### PR DESCRIPTION
**Pull request by RyanTheOptimist**
Move quic_logging_impl.{h,cc} to quiche_logging_impl.{h,cc} to match the name expected by the QUICHE platform API. Initially quic_logging_impl was the name reference from API, but in the past it changed to quiche_logging_impl. When that happened, we created quiche_logging_impl.h which simply #included quic_logging_impl.h and quic_logging_impl.h was modified to define QUICHE_foo instead of QUIC_foo. But this adds an unnecessary layer of indirection.

Risk Level: N/A
Testing: N/A
Docs Changes: N/A
Release Notes: N/A
Platform Specific Features: N/A

Signed-off-by: Ryan Hamilton <rch@google.com>

### CODE REVIEW

**Comment by daixiang0** [comment]
Please fix CI.

File: source/common/quic/platform/quiche_logging_impl.h [hunk 1]
```diff
@@ -6,4 +6,267 @@
...
+#define QUICHE_DCHECK_LT_IMPL(a, b) QUICHE_CHECK_LT_IMPL(a, b)
+#define QUICHE_DCHECK_LE_IMPL(a, b) QUICHE_CHECK_LE_IMPL(a, b)
+#define QUICHE_DCHECK_NE_IMPL(a, b) QUICHE_CHECK_NE_IMPL(a, b)
+#define QUICHE_DCHECK_EQ_IMPL(a, b) QUICHE_CHECK_EQ_IMPL(a, b)
+#endif
+
+#define QUICHE_PREDICT_FALSE_IMPL(x) ABSL_PREDICT_FALSE(x)
+#define QUICHE_PREDICT_TRUE_IMPL(x) ABSL_PREDICT_TRUE(x)
+
+namespace quic {
```
Comment (danzh2010) [inline source/common/quic/platform/quiche_logging_impl.h:133]: s/quic/quiche
Is this file moved from source/common/quic/platform/quic_logging_impl.h? Using git mv would have show the diff.

File: source/common/quic/platform/quiche_logging_impl.h [hunk 2]
```diff
@@ -6,4 +6,267 @@
...
+  static const char* guard(const char* v) { return v ? v : "(null)"; }
+};
+template <> struct NullGuard<const char*> {
+  static const char* guard(const char* v) { return v ? v : "(null)"; }
+};
+template <> struct NullGuard<std::nullptr_t> {
+  static const char* guard(const std::nullptr_t&) { return "(null)"; }
+};
+
+class QuicLogEmitter {
```
Comment (danzh2010) [inline source/common/quic/platform/quiche_logging_impl.h:175]: s/Quic/Quiche

**Review by danzh2010 — COMMENTED** [review]
Thanks for cleaning up this!

File: source/common/quic/platform/quiche_logging_impl.h (hunk 1 above)
Comment (RyanTheOptimist) [reply source/common/quic/platform/quiche_logging_impl.h]: Nice catch about "quic". There were quite a few other such terms and I renamed those too, now.

I did use git mv, but I think it's complicated by the fact that tip of tree already had a quiche_logging_impl.h, but it was just #include quic_logging_impl.h, as I mentioned in the description. I think this is preventing the move detection from working correctly :(

File: source/common/quic/platform/quiche_logging_impl.h (hunk 2 above)
Comment (RyanTheOptimist) [reply source/common/quic/platform/quiche_logging_impl.h]: Done.

**Review by RyanTheOptimist — COMMENTED** [review]
Thanks for the review!

**Review by alyssawilk — DISMISSED** [review]
s/LGTM/LACBIGINTYFD/

(Looks Annoyingly Complex But I Guess It's Necessary, Thank You For Doing it)

**Comment by RyanTheOptimist** [comment]
@alyssawilk can you re-approve?

**Review by alyssawilk — APPROVED** [review]

**Pull request merged**